States document (#3)
IamMaxim/vigilant-palm-tree: documents/interaction-states.md

# Interaction state machine

> During his work, a user can perform different actions, such as moving the mouse, leaving the seat, or typing on the keyboard. Due to the fact that those actions are immediate, it is possible to instantly transition from any state to any other.

Reviewer: ```suggestion
During their work, a user can perform different actions, such as moving the mouse, leaving the seat, or typing on the keyboard. Due to the fact that those actions are immediate, it is possible to instantly transition from any state to any other.
```
Gender neutrality

Each of the following states is defined in terms of those 3 parameters:

> - **Gaze**: workspace(💻), elsewhere (🌲), none (❌);

Reviewer: **Gaze: none** as in the person is not visible on camera? I don't think there's such a thing as "no gaze", maybe we could rename this parameter as "physical presence" with options "looking at the screen", "looking elsewhere", "not present"?

Author: Yes, by **none** I meant that person is no visible on the camera. I agree that your wording is clearer.

- **Voice**: any (❔), present (✔), none (❌);

> - **Input**: any(❔), mouse & keyboard (✔), none (❌).

Reviewer: I don't think we should be taking into consideration the input from mouse/keyboard, since we're trying to predict engagement from audio/video and **then** correlate that engagement to the keyboard/mouse activity.

Author: That was a major source of confusion for me, thanks.

## States

### Engagement

| Gaze | Voice | Input |
| ---- | ----- | ----- |
| 💻   | ❔    | ✔     |

> In this state the user is actively engaged in his work through visual focus on the workspace (that is, the monitor or the table in front of him) and active input from mouse and/or keyboard.

Reviewer: ```suggestion
In this state the user is actively engaged in their work through visual focus on the workspace (that is, the monitor or the table in front of them) and active input from the mouse and/or keyboard.
```

### Idling

| Gaze | Voice | Input |
| ---- | ----- | ----- |
| 💻   | ❔    | ❌    |

> In this state the user is passively engaged in the work, i.e. he is thinking about the problem or sketching on a paper. Due to the lack of the input, no traceable progress can be recognized.

Reviewer: ```suggestion
In this state the user is passively engaged in the work, i.e. they are thinking about the problem or sketching on paper. Due to the lack of the input, no traceable progress can be recognized.
```

### Conferencing

| Gaze | Voice | Input |
| ---- | ----- | ----- |

> | 💻   | ✔     | ❌    |

Reviewer: This state is intersecting with the **Idling** state, since it has "any" for voice. We should make them distinct.

Author: Adjusted.

> In this state the user is communicating using his voice through an app (i.e. Zoom) on his device.

Reviewer: ```suggestion
In this state the user is communicating with their voice through an app (i.e. Zoom) on their device.
```

### Distraction

| Gaze | Voice | Input |
| ---- | ----- | ----- |
| 🌲   | ❔    | ❔    |

> In this state the user is being distracted from his work by looking outside of the workspace. For example, he is talking to a person nearby.

Reviewer: ```suggestion
In this state the user is being distracted from their work by looking outside of the workspace. For example, they are talking to a person nearby.
```

### Absense

| Gaze | Voice | Input |
| ---- | ----- | ----- |
| ❌   | ❔    | ❔    |

> In this state the user has left his workplace. No work is being done.

Reviewer: ```suggestion
In this state the user has left their workplace. No work is being done.
```
